src/thinblock.h

```diff
@@ -222,6 +222,7 @@ class CThinBlockData
 extern CThinBlockData thindata; // Singleton class
 
 
+bool ReconstructBlock(CNode *pfrom, const bool fXVal, int &missingCount, int &unnecessaryCount);
```

Comment:
Let's not pollute the header file. Make it static and either predeclare in the cpp file or move the function up.